orchestration/src/test/java/com/sap/ai/sdk/orchestration/OrchestrationUnitTest.java

```diff
@@ -235,33 +233,21 @@ void filteringStrict() {
         post(urlPathEqualTo("/v2/inference/deployments/abcdef0123456789/completion"))
             .willReturn(jsonResponse(response, SC_BAD_REQUEST)));
 
-    final var filter = createAzureContentFilter(NUMBER_0);
+    final var filter =
+        new AzureContentFilter()
+            .hate(ALLOW_SAFE)
+            .selfHarm(ALLOW_SAFE)
+            .sexual(ALLOW_SAFE)
+            .violence(ALLOW_SAFE);
 
-    final var configWithFilter = config.withFilteringConfig(filter);
+    final var configWithFilter = config.withInputFiltering(filter).withOutputFiltering(filter);
```

Comment:
It would be nice to unit test the immutability, e.g. by asserting on the old and new config objects.

Could also be done in a separate, dedicated unit test class, to not increase this test size further